### PR TITLE
feat: audio visualizer.

### PR DESCRIPTION
**Pull request by cloudwebrtc**

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -38,8 +38,9 @@ class MyApp extends StatelessWidget {
 class MyHomePage extends StatelessWidget {
   const MyHomePage({super.key});
 
-  final url = 'wss://livekit.example.com';
-  final token = 'your_token_here';
+  final url = 'ws://localhost:7880';
+  final token =
+      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3MzQ3NDE4ODgsImlzcyI6IkFQSXJramtRYVZRSjVERSIsIm5hbWUiOiJtYWMiLCJuYmYiOjE3MzI5NDE4ODgsInN1YiI6Im1hYyIsInZpZGVvIjp7ImNhblVwZGF0ZU93bk1ldGFkYXRhIjp0cnVlLCJyb29tIjoibGl2ZSIsInJvb21Kb2luIjp0cnVlfX0.UzCvXREqMJxrkIsnjBsVCY78oA03flEaolNlDL9IGws';
```
Comment (bcherry) [inline example/lib/main.dart:43]: put these back?

File: example/lib/main.dart [hunk 2]
```diff
@@ -48,9 +49,9 @@ class MyHomePage extends StatelessWidget {
       print('Joining room: name=$name, roomName=$roomName');
     }
     try {
-      final details = await fetchConnectionDetails(name, roomName);
-      await roomCtx.connect(
-          url: details.serverUrl, token: details.participantToken);
+      //final details = await fetchConnectionDetails(name, roomName);
+      //await roomCtx.connect(
+      //    url: details.serverUrl, token: details.participantToken);
```
Comment (bcherry) [inline example/lib/main.dart:54]: was this intentionally removed?

File: lib/src/context/room_context.dart [hunk 3]
```diff
@@ -198,6 +199,9 @@ class RoomContext extends ChangeNotifier with ChatContextMixin {
   /// Get the [Room] instance.
   Room get room => _room;
 
+  ///
+  final bool enableAudioVisulizer;
```
Comment (bcherry) [inline lib/src/context/room_context.dart:203]: fill in the comment here - and make sure to cover the comments i made in https://github.com/livekit/client-sdk-flutter/pull/638#discussion_r1867066294

File: pubspec.yaml [hunk 4]
```diff
@@ -12,11 +12,12 @@ dependencies:
...
     sdk: flutter
   flutter_background: ^1.3.0+1
-  flutter_webrtc: ^0.12.1+hotfix.1
+  flutter_webrtc: ^0.12.3
   google_fonts: ^6.2.1
   http: ^1.2.2
   intl: ^0.19.0
-  livekit_client: ^2.3.0
+  livekit_client:
+    path: ../client-sdk-flutter
```
Comment (bcherry) [inline pubspec.yaml:20]: this should be undone, right?

**Review by bcherry — APPROVED** [review]

File: pubspec.yaml (hunk 4 above)
Comment (cloudwebrtc) [reply pubspec.yaml:20]: yeah, when client-sdk is released to pub, I can update this

File: example/lib/main.dart (hunk 2 above)
Comment (cloudwebrtc) [reply example/lib/main.dart]: This is the code for fetching token from meet.staging.livekit.io. Remove it for now and consider adding it when building the example web app later.

**Pull request merged**